### PR TITLE
Initial MLP Speculative Decoding implementation

### PR DESCRIPTION
**Pull request by christinaexyou**
**NOTE**: This is currently a draft since I have a question 

This PR addresses #41 by adding:
* `MLPSpeculatorConfig` 
* `MLPSpeculator` 
*  Relevant unit tests
---
### Questions
* Currently the `forward` method generates logits for all possible candidate sequences, should I be choosing the top candidates here through Tree Attention ?

### CODE REVIEW

**Comment by dsikka** [comment]
Hi @christinaexyou 

Has this been tested? Are you still interested in getting this landed?

We will be looking into MTP fine tuning if you’d like to help with that instead?

**Comment by dsikka** [comment]
@mergifyio refresh

**Comment by fynnsu** [comment]
Going to close this as I don't think we're going to merge MLP and the bots are going crazy
